databases/week3/mealsharing.sql

```diff
@@ -0,0 +1,678 @@
+  );
+insert into
+  reservation (
+    `id`,
+    `number_of_guests`,
+    `meal_id`,
+    `created_date`,
+    `contact_phonenumber`,
+    `contact_name`,
+    `contact_email`
+  )
+values
+  (
+    7,
+    2,
+    7,
+    "2022-03-03",
+    '648988924',
+    'René Rasmussen',
+    'rren@hotmail.com'
+  );
+insert into
+  reservation (
+    `id`,
+    `number_of_guests`,
+    `meal_id`,
+    `created_date`,
+    `contact_phonenumber`,
+    `contact_name`,
+    `contact_email`
+  )
+values
+  (
+    8,
+    8,
+    8,
+    "2022-02-14",
+    '388988924',
+    'Genella Pinto',
+    'gnlpto@hotmail.com'
+  );
+-- REVIEW
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    1,
+    'Dedang rendang',
+    "Its one of the best side dishes I tried in Malaysia",
+    1,
+    7,
+    "2022-03-03"
+  );
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    2,
+    'Paratha',
+    'A freshly made Paratha which made us feel as we are back HOME, Rating it 5 cause it felt as dunked in oil',
+    2,
+    5,
+    "2022-02-13"
+  );
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    3,
+    'Classic pizzetta',
+    'Home made italian & Mediterranean classic Pizza, it sound as it is described',
+    3,
+    8,
+    "2022-02-09"
+  );
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    4,
+    'Shawarma',
+    'Love the smell and the presentation of the food, for sure will visit again',
+    4,
+    9,
+    "2022-03-03"
+  );
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    5,
+    'Mezza Plates',
+    'Though the portion was comperatively smaller than expected it did taste super good',
+    5,
+    8,
+    "2022-03-03"
+  );
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    6,
+    'XXX- Large Tallerken (8personer)',
+    'It was a great service at the restaurant, but the sorrounding was bit noisy',
+    6,
+    9,
+    "2022-02-01"
+  );
+insert into
+  review (
+    `id`,
+    `title`,
+    `description`,
+    `meal_id`,
+    `stars`,
+    `created_date`
+  )
+values
+  (
+    7,
+    'Rød grød med fløde',
+    "Det smager det samme som min mor laver",
+    7,
+    9,
+    "2022-03-03"
+  );
+SELECT
+  *
+FROM
+  review;
+--Get meals that has a price smaller than a specific price fx 90
+SELECT
+  *
+FROM
+  meal
+WHERE
+  price < 90;
+--Get meals that still has available reservations
+SELECT
+  *
+FROM
+  meal
+  JOIN reservation ON meal.id = reservation.meal_id
+WHERE
+  reservation.id = NULL;
```

Comment:
What kind of meals do you expect to get from here?
Also the id in the reservation table is a primary key so it can never be null.

Btw, this is the trap query. You need to sum the number_of_guests over all reservations for a given meal and then compare that number with the max_reservations of that meal.

There are at least two ways of doing this, both involve stuff that we didnt talk about in class:
 - use the COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce
 - use GROUP BY, SUM, and `HAVING: https://www.w3schools.com/sql/sql_having.asp

Also remember that there might be meals that don't have any reservations yet.